src/OpenDB/src/db/dbMaster.cpp

```diff
@@ -772,30 +772,41 @@ dbMaster* dbMaster::getMaster(dbLib* lib_, uint dbid_)
 bool dbMaster::isFiller()
 {
   _dbMaster* master = (_dbMaster*) this;
-  // dbMasterType type= dbMasterType(master->_flags._type);
-
-  if (getMTermCount() == 2) {
-    bool signal = false;
-    dbSet<dbMTerm>::iterator itr;
-    dbSet<dbMTerm> mterms = getMTerms();
-    for (itr = mterms.begin(); itr != mterms.end(); ++itr) {
-      dbMTerm* mt = *itr;
-      if (!((mt->getSigType() == dbSigType::GROUND)
-            || (mt->getSigType() == dbSigType::POWER))) {
-        signal = true;
-        break;
-      }
-    }
-    if (!signal)
-      return true;
-  }
 
   switch (master->_flags._type) {
     case dbMasterType::CORE_SPACER:
       return true;
-    default:
+    case dbMasterType::CORE:
+    case dbMasterType::CORE_FEEDTHRU:
+    case dbMasterType::CORE_TIEHIGH:
+    case dbMasterType::CORE_TIELOW:
+    case dbMasterType::CORE_WELLTAP:
+    case dbMasterType::CORE_ANTENNACELL:
+    case dbMasterType::BLOCK:
+    case dbMasterType::BLOCK_BLACKBOX:
+    case dbMasterType::BLOCK_SOFT:
+    case dbMasterType::ENDCAP:
+    case dbMasterType::ENDCAP_PRE:
+    case dbMasterType::ENDCAP_POST:
+    case dbMasterType::ENDCAP_TOPLEFT:
+    case dbMasterType::ENDCAP_TOPRIGHT:
+    case dbMasterType::ENDCAP_BOTTOMLEFT:
+    case dbMasterType::ENDCAP_BOTTOMRIGHT:
+    case dbMasterType::COVER:
+    case dbMasterType::COVER_BUMP:
+    case dbMasterType::RING:
+    case dbMasterType::PAD:
+    case dbMasterType::PAD_AREAIO:
+    case dbMasterType::PAD_INPUT:
+    case dbMasterType::PAD_OUTPUT:
+    case dbMasterType::PAD_INOUT:
+    case dbMasterType::PAD_POWER:
+    case dbMasterType::PAD_SPACER:
+    case dbMasterType::NONE:
       return false;
   }
+  // gcc warning
+  return false;
```

Comment:
Shouldn't this be an assert? It seems like this is essentially just a default clause in the switch statement which seems incorrect if a new dbMasterType shows up that's considered a fill. Arguably this warning should be turned into an error so that you always have choose whether or not an enum is considered a fill

Reply:
In general warnings -> error is desirable but there is a lot of cleanup needed first.  It is on my longer term list.  Feel free to jump in :)

Reply:
asserts can be compiled out so that wouldn't be a replacement but it could be an addition.  Many other funcs are of this sort though so there is nothing new here.